test/runtests.jl

```diff
@@ -14,33 +14,24 @@ end
 
 args = Base.shell_split(get(ENV, "JULIA_MPIEXEC_TEST_ARGS", ""))
 
-function runtests()
-    nprocs = clamp(Sys.CPU_THREADS, 2, 4)
-    testdir = dirname(@__FILE__)
-    istest(f) = endswith(f, ".jl") && startswith(f, "test_")
-    testfiles = sort(filter(istest, readdir(testdir)))
+nprocs = clamp(Sys.CPU_THREADS, 2, 4)
+testdir = @__DIR__
+istest(f) = endswith(f, ".jl") && startswith(f, "test_")
+testfiles = sort(filter(istest, readdir(testdir)))
 
-    nfail = 0
-    printstyled("Running MPI.jl tests\n"; color=:white)
-    
-    for f in testfiles
-        mpiexec() do cmd
-            cmd = `$cmd $args`
-            if f == "test_spawn.jl"
-                run(`$cmd -n 1 $(Base.julia_cmd()) $(joinpath(testdir, f))`)
-            elseif f == "test_threads.jl"
-                withenv("JULIA_NUM_THREAD" => "4") do
-                    run(`$cmd -n $nprocs $(Base.julia_cmd()) $(joinpath(testdir, f))`)
-                end
-            else
-                run(`$cmd -n $nprocs $(Base.julia_cmd()) $(joinpath(testdir, f))`)
+@testset "$f" for f in testfiles
```

Comment:
is the `@testset` worth it? each one only has one test.

Reply:
It's just a way to get the names of the tests printed in a consistent fashion using the existing testing framework

Reply:
fair enough.